experimental/ir/lower_resolve.go

```diff
@@ -144,6 +148,45 @@ func resolveExtendeeType(c *Context, extendee *rawExtendee, r *report.Report) {
 	}
 }
 
+func resolveLangSymbols(c *Context) {
+	if !c.isDescriptorProto {
+		return
+	}
+
+	field := func(name string) arena.Pointer[rawField] {
+		return mustResolve[rawField](c, "google.protobuf."+name, SymbolKindField)
+	}
+
+	l := inferNew(&c.langSymbols)
+
+	l.fileOptions = field("FileDescriptorProto.options")
+
+	l.messageOptions = field("DescriptorProto.options")
+	l.fieldOptions = field("FieldDescriptorProto.options")
+	l.oneofOptions = field("OneofDescriptorProto.options")
+
+	l.enumOptions = field("EnumDescriptorProto.options")
+	l.enumValueOptions = field("EnumValueDescriptorProto.options")
+}
+
+// inferNew is a helper for allocating a value of a large, complicated type.
+// Specifically, it's used with Context.langSymbols.
+func inferNew[T any](p **T) *T {
+	*p = new(T)
+	return *p
+}
```

Comment:
This is kinda confusing and seems totally unnecessary. Why not just create an unexported _named_ type for `langSymbols` instead?